external_executors/dnanexus_wes/README.md

```diff
@@ -0,0 +1,73 @@
+# dxWES
+
+## Summary
+
+This is a cloudformation stack you can run on AWS. It provides several
+[WES](https://github.com/ga4gh/workflow-execution-schemas) API endpoints and forwards those
+requests to the DNAnexus platform.
+
+Currently, three WES endpoints are implements:
+
+- POST `/workflows`
+- GET `/workflows/{workflow_id}/status`
+- GET `/workflows/{workflow_id}`
+
+## Deploying on AWS
+
+There is a deployment script that can be run with
+
+```
+sh deploy.sh <stack_name> <bucket_name>
+```
+
+You can do it by hand:
+
+### Prepare the lambda deployment package
+
+The `dx-wes-lambda` directory needs some additional resources before it can be
+zipped up and uploaded to S3:
+
+1. The dxWDL jar, available [here](https://github.com/dnanexus/dxWDL/releases)
+2. dx-toolkit python dependencies.
+    a. Download the dx-toolkit tarball for Ubuntu 14.04 from
+[DNAnexus](https://wiki.dnanexus.com/downloads)
+    b. Extract the tarball
+    c. Copy everything in `dx-toolkit/share/dnanexus/lib/python2.7/site-packages/` into
+`dx-wes-lambda`
+3. dx-toolkit executables
+    a. Make a `dx-wes-lambda/bin` directory.
+    b. Copy all the files from dx-toolkit/bin/ into `dx-wes-lambda/bin`
+
+Now zip the directory and upload it to S3:
+```
+cd dx-wes-lambda && zip -X -r ../dx_wes.zip * && cd ..
+aws s3 cp dx_wes.zip s3://my-bucket-name/dx_wes.zip
+```
+
+### Create the Cloudformation Stack
+
+You can create the stack via the CLI using the JSON template:
+```
+aws cloudformation create-stack --stack-name "dx-wes" \
+    --template-body file://dx_wes.json \
+    --parameters ParameterKey=LambdaCodeBucket,ParameterValue=my-bucket-name \
+                 ParameterKey=LambdaCodeKey,ParameterValue=dx_wes.zip \
+    --capabilities CAPABILITY_NAMED_IAM
+```
+
+## Test the API
+
+Create a JSON with the workflow request that looks like this:
+
+```
```

Comment:
Is there a way to make this look less repulsively formatted in the README? :)

Reply:
I wish, I think it needs to be a great big string with no newlines.

Reply:
:+1: